Add RPM packaging

mwperina opened the pull request:
- Add new document describing the expected DBUS apis
- Revert accidental rename of "Node" section name in config.
- Complete ListUnit end-to-end handling
- Add RPM packaging

mwperina committed:
Add RPM packaging

Adds RPM packages for the main orchestrator and for the agent.

Signed-off-by: Martin Perina <mperina@redhat.com>

sdunnagan commented on hirte.spec.in:
```diff
@@ -0,0 +1,57 @@
+Name:		hirte
+Version:	@VERSION@
+Release:	@RELEASE@%{?dist}
+Summary:	Container orchestrator for edge based clusters
```
How about, "Hirte is a tool for orchestrating systemd services on remote machines."

mwperina replied:
Just now aligned with suggestions from https://github.com/containers/hirte/pull/97

sdunnagan commented on hirte.spec.in:
```diff
@@ -0,0 +1,57 @@
+
+
+%description
+This package contains the Hirte container orchestrator for edge based clusters.
```
"Hirte is a tool for orchestrating systemd services on remote machines."

mwperina replied:
Just now aligned with suggestions from https://github.com/containers/hirte/pull/97

sdunnagan commented on hirte.spec.in:
```diff
@@ -0,0 +1,57 @@
+
+
+%description agent
+This packages contains the agent for Hirte container orchestrator.
```
"This package contains the agent for Hirte."

mwperina replied:
Just now aligned with comments from https://github.com/containers/hirte/pull/97

mwperina committed:
Fix unused return value in parse-util_test.c

We have enabled treating warnings as errors in the project, so RPM build
fails on not using result code of asprintf function inside
parse-util.test.c. This patch assings the return value to a variable to
bypass that error.

Signed-off-by: Martin Perina <mperina@redhat.com>

alexlarsson commented on hirte.spec.in:
```diff
@@ -0,0 +1,72 @@
+URL:		https://github.com/containers/hirte
+Source0:	%{name}-%{version}.tar.gz
+
+BuildArch:	x86_64 aarch64
```
I don't think we should list these only. There is nothing arch-specific about hirte and it should build everywhere if you want.

mwperina replied:
Done
